src/pandas_profiling/__init__.py

```diff
@@ -271,3 +330,146 @@ def to_app(self):
         app_widgets = QtReport(self.report).render()
 
         app = get_app(app, self.title, app_widgets)
+
+    def _repr_html_(self):
+        """The ipython notebook widgets user interface gets called by the jupyter notebook."""
+        self.to_notebook_iframe()
+
+    def __repr__(self):
+        """Override so that Jupyter Notebook does not print the object."""
+        return ""
+
+    def dumps(self) -> bytes:
```

Comment:
Style suggestion: implement the serialization features as a mixin class. 

Example for JSON:
https://gist.github.com/jimbattin/dd5cac131b582d8201d18992d52b6fdb